Scripts/Tools/SurfaceEditor.cs

```diff
@@ -915,6 +1015,50 @@ void OnRepaint (SceneView sceneView, Event e)
 				GL.End();
 				GL.PopMatrix();
 			}
+
+            // Deselect surfaces that are not hidden during "find hidden surfaces"
+            if (findingHiddenFaces)
+            {
+                List<Polygon> toDeselect = new List<Polygon>();
+                for (int polygonIndex = 0; polygonIndex < selectedSourcePolygons.Count; polygonIndex++)
+                {
+                    Polygon polygon = selectedSourcePolygons[polygonIndex];
+                    Brush brush = csgModel.FindBrushFromPolygon(polygon);
+
+                    // todo: I don't understand... are the normals simply inverted due to some math bug elsewhere? Inverting my logic depending on the CSG Mode fixes it...
```

Comment:
This is the correct logic, as you're comparing source polygons rather than built polygons you're working with the brush polygons which always point outwards regardless of if they're subtractive or not. It's the build process itself which is responsible for making the subtracted faces have the correct normal. This code looks fine as-is but it would be good to remove the TODO before merging